openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/Shared/ActivityFeedActions.tsx

```diff
@@ -11,62 +11,109 @@
+import { ReactionOperation } from '../../../enums/reactions.enum';
 import {
   Post,
+  ReactionType,
   Thread,
   ThreadType,
 } from '../../../generated/entity/feed/thread';
-
-import { ReactComponent as IconReply } from '../../../assets/svg/ic-reply.svg';
 import { useApplicationStore } from '../../../hooks/useApplicationStore';
+import {
+  deletePostById,
+  deleteThread,
+  updatePost,
+  updateThread,
+} from '../../../rest/feedsAPI';
+import { getUpdatedThread } from '../../../utils/FeedUtils';
 import { useActivityFeedProvider } from '../ActivityFeedProvider/ActivityFeedProvider';
+import Reaction from '../Reactions/Reaction';
 import './activity-feed-actions.less';
 
 interface ActivityFeedActionsProps {
   post: Post;
   feed: Thread;
   isPost: boolean;
   onEditPost?: () => void;
+  isAnnouncementTab?: boolean;
+  updateAnnouncementThreads?: () => void;
+  permissions?: boolean;
 }
 
 const ActivityFeedActions = ({
   post,
   feed,
   isPost,
   onEditPost,
+  isAnnouncementTab,
+  updateAnnouncementThreads,
+  permissions,
 }: ActivityFeedActionsProps) => {
   const { t } = useTranslation();
   const { currentUser } = useApplicationStore();
   const isAuthor = post.from === currentUser?.name;
   const [showDeleteDialog, setShowDeleteDialog] = useState(false);
-  const { deleteFeed, showDrawer, hideDrawer, updateEditorFocus } =
-    useActivityFeedProvider();
-
+  const {
+    deleteFeed,
+    showDrawer,
+    hideDrawer,
+    updateEditorFocus,
+    updateEntityThread,
+  } = useActivityFeedProvider();
+  const [visible, setVisible] = useState<boolean>(false);
   const onReply = () => {
     showDrawer(feed);
 
     updateEditorFocus(true);
   };
 
+  // delete Announcement in Service page's Announcement Tab
+  const deleteAnnouncementFeed = useCallback(
+    async (threadId: string, postId: string, isThread: boolean) => {
+      if (isThread) {
+        await deleteThread(threadId);
+      } else {
+        await deletePostById(threadId, postId);
+        await getUpdatedThread(threadId);
+      }
+      updateAnnouncementThreads && updateAnnouncementThreads();
+    },
+    []
+  );
+
   const handleDelete = () => {
-    deleteFeed(feed.id, post.id, !isPost).catch(() => {
-      // ignore since error is displayed in toast in the parent promise.
-    });
+    if (isAnnouncementTab && feed.type === ThreadType.Announcement) {
+      deleteAnnouncementFeed(feed.id, post.id, !isPost); // delete Announcement feed in Service page's Announcement Tab
+    } else {
+      deleteFeed(feed.id, post.id, !isPost).catch(() => {
+        // ignore since error is displayed in toast in the parent promise.
+      });
+    }
+
     setShowDeleteDialog(false);
     if (!isPost) {
       hideDrawer();
     }
   };
 
   const editCheck = useMemo(() => {
-    if (feed.type === ThreadType.Announcement && !isPost) {
-      return false;
+    if (feed.type === ThreadType.Announcement) {
```

Comment:
we can memoise this `feed.type === ThreadType.Announcement,` since used multiple places here!